fe/fe-core/src/main/java/org/apache/doris/qe/PointQueryExecutor.java

```diff
@@ -208,21 +409,61 @@ public RowBatch getNext() throws Exception {
         if (candidateBackends == null || candidateBackends.isEmpty()) {
             return new RowBatch();
         }
+        OlapTable olapTable = shortCircuitQueryContext.scanNode.getOlapTable();
+        List<Column> keyColumns = olapTable.getBaseSchemaKeyColumns();
+        for (int i = 0; i < keyColumns.size(); ++i) {
+            Column column = keyColumns.get(i);
+            if (olapTable.isDistributionColumn(column.getName())) {
+                distributionKeyColumns.add(i);
+            } else if (olapTable.isPartitionColumn(column.getName())) {
+                partitionKeyColumns.add(i);
+            }
+        }
         Iterator<Backend> backendIter = candidateBackends.iterator();
```

Comment:
candidateBackends contains all replicas of tablets , you should only request one of the replica